autoload/ale/c.vim

```diff
@@ -91,3 +92,27 @@ function! ale#c#FindCompileCommands(buffer) abort
 
     return ''
 endfunction
+
+" Given a buffer number, return the relevant command from
+" compile_commands.json. Return 0 if none could be found.
+function! ale#c#GetCompileCommand(buffer) abort
+    let l:build_dir = ale#c#FindCompileCommands(a:buffer)
+    if empty(l:build_dir)
+        return 0
+    endif
+    let l:commands = json_decode(readfile(l:build_dir . '/compile_commands.json'))
+    let l:file = fnamemodify(bufname(a:buffer), ':p')
+    " save current directory, so we can restore it before returning
+    let l:cwd = getcwd()
+    for l:command in l:commands
+        " temporarily change dir to run fnamemodify
+        execute 'cd' l:command.directory
```

Comment:
Don't change the working directory, as it could break other plugins that are being run. If you want want to get an absolute path for a relative path, you can use `ale#path#GetAbsPath(l:base_directory, l:filename)`.